### PR TITLE
fixed bids offers trades accumulation in the endpoint buffer

### PR DESCRIPTION
**Pull request by hannesdiedrich**

### CODE REVIEW

File: src/d3a/d3a_core/sim_results/endpoint_buffer.py [hunk 1]
```diff
@@ -312,7 +312,8 @@ def update_offer_bid_trade(self):
         if self.current_market_time_slot_str == "":
             return
         for area_uuid, area_result in self.flattened_area_core_stats_dict.items():
-            self.bids_offers_trades[area_uuid] = area_result
+            self.bids_offers_trades[area_uuid] = \
+                {k: area_result[k] for k in ('offers', 'bids', 'trades')}
```
Comment (faizan2590) [inline src/d3a/d3a_core/sim_results/endpoint_buffer.py:316]: One more thing would be nice to have is to sanitise `offers', 'bids', 'trades'` i.e there is no need to send the complete serialised version of items contained in these objects

**Review by faizan2590 — APPROVED** [review]
One enhancement comments, other than that LGTM

File: src/d3a/d3a_core/sim_results/endpoint_buffer.py (hunk 1 above)
Comment (hannesdiedrich) [reply src/d3a/d3a_core/sim_results/endpoint_buffer.py:316]: @faizan2590 You mean by only sending the bids/offers/trades that belong to the area? it hink this is already done by design of `flattened_area_core_stats_dict` or am i misunderstanding?

Comment (hannesdiedrich) [reply src/d3a/d3a_core/sim_results/endpoint_buffer.py:316]: Discussed offline: Only sanitise the offer bid and trades dicts from data that is not needed to be stored in the DB when research team decided which data is valueable or not.

**Pull request merged**